include/SDL3/SDL_iostream.h

```diff
@@ -133,6 +133,17 @@ typedef struct SDL_IOStreamInterface
      */
     size_t (SDLCALL *write)(void *userdata, const void *ptr, size_t size, SDL_IOStatus *status);
 
+    /**
+     *  If the stream is buffering, make sure the data is written out.
+     *
+     *  On failure, you should set `*status` to a value from the
+     *  SDL_IOStatus enum. You do not have to explicitly set this on
+     *  a successful flush.
+     *
+     *  \return SDL_TRUE if successful or SDL_FALSE on write error when flushing data.
+     */
+    SDL_bool (SDLCALL *flush)(void *userdata, SDL_IOStatus *status);
```

Comment:
If we're going to need this, I think it might make sense to land it as its own feature ahead of SDL_Process. Adding it to the vtable is an ABI break if I'm reading correctly, so this needs to land before the ABI freeze milestone even if SDL_Process isn't ready.

Reply:
Yep, I can break this out if it looks like this PR isn't landing before the ABI break for some reason.